lib/dynflow/action/rescue.rb

```diff
@@ -44,14 +44,20 @@ def rescue_strategy_for_planned_action(action)
       action.rescue_strategy
     end
 
-    # Override when different appraoch should be taken for combining
+    # Override when different approach should be taken for combining
     # the suggested strategies
     def combine_suggested_strategies(suggested_strategies)
-      if suggested_strategies.empty? ||
-            suggested_strategies.all? { |suggested_strategy| suggested_strategy.strategy == Skip }
+      if suggested_strategies.empty?
         return Skip
       else
-        return Pause
+        # TODO: Find the safest rescue strategy among the suggested ones
+        if suggested_strategies.all? { |suggested_strategy| suggested_strategy.strategy == Skip }
+          return Skip
+        elsif suggested_strategies.any? { |suggested_strategy| suggested_strategy.strategy == Fail }
+          return Fail
```

Comment:
I'm not sure if one sub-action somewhere at the bottom of the plan tree should be able to set the whole task as failed. Notice that in Skip, all the sub-plans need to agree on the skip before proceeding, Also, the fail strategy is influencing also the steps that were not started yet. I think, for this strategy, should win only if all the pending steps are planned by the action that suggests the Fail strategy.